### ISSUE TITLE
`chaperone-evt` on a will executor fails on racketcs

### ISSUE THREAD

**Issue by samth** [racket/racket#2730]
```
[samth@huor:~/sw/plt/extra-pkgs/typed-racket (master) plt] racketcs
Welcome to Racket v7.4.0.1 [cs].
> (chaperone-evt (make-will-executor) (lambda (e) (values e values)))
; chaperone-struct: cannot chaperone instance of an authentic structure type
;   given value: #<will-executor>

[samth@huor:~/sw/plt/extra-pkgs/typed-racket (master) plt] racket
Welcome to Racket v7.4.0.1.
> (chaperone-evt (make-will-executor) (lambda (e) (values e values)))
#<will-executor>
```

**Comment by samth** [comment]
I think this also happens for `tcp-listener` structs, probably for the same reason. 

To avoid too many round-trips here, you might want to run `typed-racket-test/succeed/pr241-variation-5.rkt` and see if it works successfully.

**Comment by mflatt** [comment]
Fixed by 997f5b583c8030e089d87ade943be524774f2ad0?

**Comment by sorawee** [comment]
Ran `typed-racket-test/succeed/pr241-variation-5.rkt` and it seems to work fine. Closing.
